modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/AppContext.es.js

```diff
@@ -235,7 +235,7 @@ const createReducer = dataLayoutBuilder => {
 					delete dataRule['logical-operator'];
 				}
 
-				dataRule.id = new Date().getTime();
+				dataRule.ruleId = dataRules.length + 1;
```

Comment:
should be probably 

dataRule.ruleId = dataRules.length (without + 1), ruleIds should start at 0, otherwise there will be too many conversions between their data array index and ids.